CONTRIBUTING.md

```diff
@@ -0,0 +1,222 @@
+
+## Pull requests
+
+Good pull requests - patches, improvements, new features - are a fantastic
+help. They should remain focused in scope and avoid containing unrelated
+commits.
+
+**IMPORTANT**: By submitting a patch, you agree that your work will be
+licensed under the license used by the project.
+
+If you have any large pull request in mind (e.g. implementing features,
+refactoring code, etc), **please ask first** otherwise you risk spending
+a lot of time working on something that the project's developers might
+not want to merge into the project.
+
+Please adhere to the coding conventions in the project (indentation,
+accurate comments, etc.) and don't forget to add your own tests and
+documentation. When working with Git, we recommend the following process
+in order to craft an excellent pull request:
+
+1. [Fork](https://help.github.com/fork-a-repo/) the project, clone your fork,
+  and configure the remotes:
+
+  ```sh
+  # Clone your fork of the repo into the current directory
+  git clone https://github.com/<your-username>/elixir-nats
+  # Navigate to the newly cloned directory
+  cd elixir-nats
+  # Assign the original repo to a remote called "upstream"
+  git remote add upstream https://github.com/nats-io/elixir-nats
+  ```
+
+2. If you cloned a while ago, get the latest changes from upstream:
+
+  ```sh
+  git checkout master
+  git pull upstream master
+  ```
+
+3. Create a new topic branch (off of `master`) to contain your feature, change,
+  or fix.
+
+  **IMPORTANT**: Making changes in `master` is discouraged. You should always
+  keep your local `master` in sync with upstream `master` and make your
+  changes in topic branches.
+
+  ```sh
+  git checkout -b <topic-branch-name>
+  ```
+
+4. Commit your changes in logical chunks. Keep your commit messages organized,
+  with a short description in the first line and more detailed information on
+  the following lines. Feel free to use Git's
+  [interactive rebase](https://help.github.com/articles/interactive-rebase)
+  feature to tidy up your commits before making them public.
+
+5. Make sure all the tests are still passing.
+
+  ```sh
+  mix test
+  ```
+
+  This command will compile the code in your branch and use that
+  version of Elixir to run the tests. This is needed to ensure your changes can
+  pass all the tests.
+
+6. Push your topic branch up to your fork:
+
+  ```sh
+  git push origin <topic-branch-name>
+  ```
+
+7. [Open a Pull Request](https://help.github.com/articles/using-pull-requests/)
+  with a clear title and description.
+
+8. If you haven't updated your pull request for a while, you should consider
+  rebasing on master and resolving any conflicts.
+
+  **IMPORTANT**: _Never ever_ merge upstream `master` into your branches. You
+  should always `git rebase` on `master` to bring your changes up to date when
+  necessary.
+
+  ```sh
+  git checkout master
+  git pull upstream master
+  git checkout <your-topic-branch>
+  git rebase master
+  ```
+
```

Comment:
I'm hoping like the section above we can refer by link/reference vs. copy. Thoughts?